### ISSUE TITLE
Don't show trackers for uninstalled apps

### ISSUE THREAD

**Issue by Hanouta** [Exodus-Privacy/exodus-android-app#302]
Currently apps are not being removed from the list after they have been uninstalled.

**Comment by Jean-BaptisteC** [comment]
Yes you right, the best way is to remove apps and reports uninstalled when users actualize reports

**Comment by tarunsamanta2k20** [comment]
I'd like to look into this.

**Comment by Jean-BaptisteC** [comment]
Yes feel free open PR, don't ask to work on issue

**Comment by tarunsamanta2k20** [comment]
@Jean-BaptisteC how to determine if an app has been deleted?

**Comment by tarunsamanta2k20** [comment]
@Jean-BaptisteC should i again fetch and compare?

**Comment by tarunsamanta2k20** [comment]
@Jean-BaptisteC i have prepared the DAO and service function. now from where should i call the service function?

Should I suspend the service function?

**Comment by Jean-BaptisteC** [comment]
To determine if app is deleted, we can use getApplicationList and see if app if present in list or not.
Could you compare when service call getApplicationList to doing one call and not increase refresh time ?

**Comment by tarunsamanta2k20** [comment]
> 

No for knowing which app has been deleted, I have to compare the updated list with existing list. If not present then I will add those apps to list and delete those exodusapp from db. This is my approach. Now one problem I am facing at which point should I call the service method? 


Could you please let me know what are you thinking or any other approach?

**Comment by tarunsamanta2k20** [comment]
It would be good if you please write the steps over here.

**Comment by Jean-BaptisteC** [comment]
- Start Exodus service
-  call getApplicationList
- Update db and remove uninstalled apps
- Get reports and trackers from Exodus server

**Comment by tarunsamanta2k20** [comment]
> * Update db and remove uninstalled apps

AfterUninstall getApplicationlist will give the same result right? How do I get the uninstalled apps?

**Comment by tarunsamanta2k20** [comment]
@Jean-BaptisteC i haven't completed the task yet. Should I push till the point i did? then you may mention the exact location where I have to make the changes.

**Comment by Jean-BaptisteC** [comment]
- To get uninstalled apps, you can compare result of getApplicationList function and `SELECT packageName FROM exodusapplication`
- Create PR when you want, i'm not an expert development, just try to following structure of app
